src/dotnet/UI.Blazor/Components/VisualMediaViewerModal/visual-media-viewer.ts

```diff
@@ -212,6 +219,50 @@ export class VisualMediaViewer {
         spinner.remove();
     }
 
+    private videoPlugHandler(video: HTMLMediaElement) {
+        const wrapper = video.closest('.video-wrapper') as HTMLElement;
+        if (!wrapper)
+            return;
+
+        const thumbnailWrapper = wrapper.querySelector('.video-thumbnail-wrapper') as HTMLElement;
+        if (!thumbnailWrapper)
+            return;
+
+        const thumbnail = wrapper.querySelector('.video-thumbnail') as HTMLElement;
+        if (!thumbnail)
+            return;
+
+        const spinner = wrapper.querySelector('.spinner-icon-wrapper') as HTMLElement;
+
+        if (video.readyState == 4) {
+            thumbnailWrapper.remove();
+            if (spinner)
+                spinner.remove();
+            return;
+        }
+
+        const imagePlug = thumbnail.querySelector('image-skeleton') as HTMLImageElement;
+        if (imagePlug) {
+            let plugWidth = 0;
+            let plugHeight = 0;
+            const originalWidth = thumbnail.getAttribute('data-width') as Number;
+            const originalHeight = thumbnail.getAttribute('data-height') as Number;
```

Comment:
usually it's accessed via dataset
https://developer.mozilla.org/en-US/docs/Learn_web_development/Howto/Solve_HTML_problems/Use_data_attributes#javascript_access